src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,103 @@
+package duke;
+
+import java.util.Scanner;
+
+/**
+ * Class containing the main Duke application.
+ * Initialises the application.
+ */
+public class Duke {
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        Storage st = new Storage("./data/dukedata.txt");
+        TaskList tl = new TaskList(st.setFile());
+        Ui ui = new Ui();
+
+        ui.printIntro();
+
+        while (true) {
+            String curr = sc.nextLine();
+            String[] msgWords = curr.split("\\b");
+            for (int i = 0; i < msgWords.length; i++) {
+                if (msgWords[i].equalsIgnoreCase("bye")) {
+                    sc.close();
+                    ui.goodByeMessage();
+                    return;
+                } else if (msgWords[i].equalsIgnoreCase("list")) {
+                    ui.printList(tl);
+                    break;
+                } else if (msgWords[i].equalsIgnoreCase("mark")) {
+                    int index = Integer.parseInt(msgWords[2]) - 1;
+                    if (index < 1 || index > tl.size()) {
```

Comment:
Try to avoid deep nesting (more than 3 indentations), perhaps use switch to make your code cleaner :)